migrations/20200422144952_structure_project_version_continuation_data.js

```diff
@@ -0,0 +1,156 @@
+    const matched = text.match(EXPIRYLONG2);
+    const day = matched[25];
+    const month = matched[1];
+    const year = matched[29];
+    date = moment(`${day} ${month} ${year}`, 'DD MMM YY');
+  } else if (text.match(EXPIRYSHORT)) {
+    const matched = text.match(EXPIRYSHORT);
+    let day = matched[1];
+    let month = matched[3];
+    const year = matched[6];
+    if (day.length === 1) {
+      day = '0' + day;
+    }
+    if (month.length === 1) {
+      month = '0' + month;
+    }
+    date = moment(`${day} ${month} ${year}`, 'DD MM YY');
+  }
+
+  if (!ppl.length && !date) {
+    console.log(`Cannot parse versionId: ${versionId}`);
+    stringifier && stringifier.write([TYPES.NEITHER, projectId, versionId, projectStatus, versionStatus]);
+    return;
+  }
+
+  if (!ppl.length) {
+    console.log(`Unable to parse licence number, versionId: ${versionId}`);
+    stringifier && stringifier.write([TYPES.LICENCE_NUMBER, projectId, versionId, projectStatus, versionStatus]);
+  }
+
+  if (ppl.length > 1) {
+    console.log(`Multple licence numbers found for versionId: ${versionId}`);
+    stringifier && stringifier.write([TYPES.MULTIPLE, projectId, versionId, projectStatus, versionStatus]);
+  }
+
+  if (!date) {
+    console.log(`Unable to parse expiry date, versionId: ${versionId}`);
+    stringifier && stringifier.write([TYPES.DATE, projectId, versionId, projectStatus, versionStatus]);
+  }
+
+  return {
+    'project-continuation': [
+      {
+        'licence-number': get(ppl, 0, null),
+        'expiry-date': date ? date.format('YYYY-MM-DD') : null
+      }
+    ]
+  }
+};
+
+exports.transform = transform;
+
+exports.up = function(knex) {
+  const stringifier = csv();
+  stringifier.write(['Type', 'Project ID', 'Version ID', 'Project Status', 'Version Status']);
+  return Promise.resolve()
+    .then(() => {
+      return knex('project_versions')
+        .select('project_versions.id')
+        .join('projects', 'project_versions.project_id', 'projects.id')
+        .where({ 'schema_version':  1 });
+    })
+    .then(versions => {
+      console.log(`found ${versions.length} versions`)
+      return versions.reduce((promise, version, index) => {
+        return promise
+          .then(() => {
+            console.log(`patching version: ${version.id}, ${index + 1} of ${versions.length}`);
+            return knex('project_versions')
+              .join('projects', 'project_versions.project_id', 'projects.id')
+              .select('project_versions.id', 'data', 'project_versions.status', 'projects.status as projectStatus', 'project_id')
+              .where({ 'project_versions.id': version.id })
+              .first()
+              .then(version => {
+                const loggingParams = {
+                  versionId: version.id,
+                  projectId: version.project_id,
+                  versionStatus: version.status,
+                  projectStatus: version.projectStatus
+                };
```

Comment:
Can we include the title and the establishment id in any log output please?

Reply:
👍